azuredevops/internal/service/workitemtracking/resource_workitemquery.go

```diff
@@ -0,0 +1,216 @@
+				ConflictsWith: []string{"parent_id"},
+			},
+
+			"wiql": {
+				Type:     schema.TypeString,
+				Required: true,
+				// The value of 32000 matches the restrictions in Azure DevOps.
+				ValidateFunc: validation.StringLenBetween(1, 32000),
+			},
+		},
+	}
+}
+
+func resourceQueryCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	clients := m.(*client.AggregatedClient)
+
+	projectID := d.Get("project_id").(string)
+
+	// A parent ID and area cannot be set together.
+	if d.Get("area").(string) != "" && d.Get("parent_id").(string) != "" {
+		return diag.Errorf("Only one of 'area' or 'parent_id' should be set.")
+	}
+
+	// Use the area
+	parent := d.Get("area").(string)
+
+	// Or if it's not set, use the parent ID
+	if parent == "" {
+		parent = d.Get("parent_id").(string)
+	}
+
+	params := workitemtracking.CreateQueryArgs{
+		Project: &projectID,
+		Query:   &parent,
+		PostedQuery: &workitemtracking.QueryHierarchyItem{
+			Name:     converter.String(d.Get("name").(string)),
+			Wiql:     converter.String(d.Get("wiql").(string)),
+			IsFolder: converter.Bool(false),
+		},
+	}
+
+	resp, err := clients.WorkItemTrackingClient.CreateQuery(clients.Ctx, params)
+	if err != nil {
+		return diag.Errorf(" Creating query. Error: %s", err)
+	}
+
+	if resp.Id == nil {
+		return diag.Errorf(" Creating query. Error: ID was nil")
+	}
+
+	d.SetId(resp.Id.String())
+
+	return resourceQueryRead(clients.Ctx, d, m)
+}
+
+func resourceQueryRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	clients := m.(*client.AggregatedClient)
+
+	if d.Id() == "" {
+		return diag.Errorf(" Reading query: ID was not set")
+	}
+
+	id := d.Id()
+
+	params := workitemtracking.GetQueryArgs{
+		Project: converter.String(d.Get("project_id").(string)),
+		Query:   &id,
+	}
+
+	resp, err := clients.WorkItemTrackingClient.GetQuery(clients.Ctx, params)
+	if err != nil {
+		if utils.ResponseWasNotFound(err) {
+			d.SetId("")
+			return nil
+		}
+		return diag.Errorf(" Getting query with id: %s. Error: %+v", id, err)
+	}
+
+	// Check that the query is not a folder
+	if resp.IsFolder != nil && *resp.IsFolder {
+		return diag.Errorf(" The query with id: %s is a folder. Expected a query.", id)
+	}
+
+	if resp != nil {
+		if resp.Name != nil {
+			d.Set("name", resp.Name)
+		}
+		if resp.Wiql != nil {
+			d.Set("wiql", resp.Wiql)
+		}
+	}
+	return nil
+}
+
+func resourceQueryUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	clients := m.(*client.AggregatedClient)
+	var diags diag.Diagnostics
+
+	id := d.Id()
+
+	params := workitemtracking.GetQueryArgs{
+		Project: converter.String(d.Get("project_id").(string)),
+		Query:   &id,
+	}
+
+	existing, err := clients.WorkItemTrackingClient.GetQuery(clients.Ctx, params)
+	if err != nil {
+		if utils.ResponseWasNotFound(err) {
+			d.SetId("")
+			return nil
```

Comment:
During Update, if the read returns 404, we don't remove it from state, but just throw error.